aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -340,6 +330,39 @@ void persistGlueTable(Table glueTable, Map<String, String> parameters, TableMeta
     }
   }
 
+  private void handleAWSExceptions(AwsServiceException persistFailure) {
+    int statusCode = persistFailure.statusCode();
```

Comment:
Very minor nit: Generally for better readability I try and having variables as close as possible to where they are read. I think we can move the status code to the else block or have some separate helper method `shouldThrow(AwsServiceexception)` which just returns based on the critera

Reply:
I will refactor to put it as a separate check inside the else block.